### ISSUE TITLE
Auto Demo Name Format

### ISSUE THREAD

**Issue by Mako88** [dxx-rebirth/dxx-rebirth#128]
It would be nice when the "auto record demo" feature is enabled to automatically name demos as well.

The format for a 2 player game could be: date-p1name-p2name-score-<incrementing number> (it's probably rare that the number would be used).

And any more than 2 players could be: date-anarchy-<incrementing number>

**Comment by vLKp** [comment]
There is a separate option for automatically naming demos, though it does not provide the specific naming scheme you requested.  You can set the name to **.** to choose a name based on the system time, or you can set a name using %-escapes to insert time-based values and $-escapes to insert game-based values.  The %-escapes are passed to **strftime**.  Currently, the only defined $-escapes are **$m** for **Current_mission_filename** and **$p** for the callsign of the recording player.  More escapes could be added.

**Comment by Mako88** [comment]
It'd be nice to add a string $o for other player that will automatically be the name of a second player if there is only one, or the word "anarchy" if there are multiple. Also have a $s for score would be nice.

When I get a chance I'll add a comment to the .ini and --help printout explaining the naming format.

**Comment by Mako88** [comment]
One other question, can a simple string be added? So something like -autodemoformat "Hello"-$p-%l:%P ?

**Comment by vLKp** [comment]
Yes, any sequence other than a %-escape or $-escape is copied through unchanged.
